### ISSUE TITLE
Working on *just* the UX?

### ISSUE THREAD

**Issue by BigBlueHat** [apache/incubator-ponymail#468]
I'd love to contribute to the UX, but setting up Elastic Search, Python3, Lua, and Apache or NGINX is a bit much to tweak some HTML and JS which gets it's "meaning" from a JSON API.

Would it be possible to either:
a) allow the UX to work against a remote API (i.e. test UX against https://lists.apache.org/)
b) mock the API (perhaps via an API documenting format like [OpenAPI](https://www.openapis.org/), [RAML](https://raml.org/), etc.)

Option A seems the simplest/fastest (assuming the API is or could be setup to enable CORS--see http://www.test-cors.org/).

The lower the bar is for contribution, the more designer/UX types (who likely don't have Elastic Search et al handily installed and running) might be able to help this project.

Just an idea! 💡 
🎩

**Comment by Humbedooh** [comment]
You can use remote API calls, if CORS allows. I would suggest setting up a proxy at home that tweaks CORS to allow for credentials if needed, I don't think it's gonna happen on lists.a.o :)

**Comment by BigBlueHat** [comment]
@Humbedooh actually, thanks to [this infra ticket](https://issues.apache.org/jira/browse/INFRA-15587), I probably have what I need from the CORS side of things (at least for this running instance).

The next step (using option A above) would be build a simple/alternate UX that uses that API and is built as a "standalone" web app (which could run on a different server/host, but still serve the same data).

The goal was to make UX iteration simpler/faster by separating it from server-side concerns using the API as the "glue" between them. 😃 

So, I suppose this is partly possible already, and would just need a 👍 and a more "visible" (via API docs or similar) API.

**Comment by Humbedooh** [comment]
I agree, and there is some work in progress on porting it all to python, with an OpenAPI specification instead, but work is progressing very slowly on that front.

**Comment by BigBlueHat** [comment]
> I agree, and there is some work in progress on porting it all to python, with an OpenAPI specification instead, but work is progressing very slowly on that front.

Where's that work happening? If ever I find time, I might put some of it here. 😉
